### ISSUE TITLE
"java.net.ConnectException: Connection timed out" when using flutter run

### ISSUE THREAD

**Issue by pavariss** [flutter/flutter#33170]
```
PS D:\MySpace\Flutter\FlutterProject\myapp1> flutter run
Using hardware rendering with device Android SDK built for x86. If you get graphics artifacts, consider enabling software rendering with
"--enable-software-rendering".
Launching lib/main.dart on Android SDK built for x86 in debug mode...
Initializing gradle...                                                 -
Oops; flutter has exited unexpectedly.                                       Sending crash report to Google.
Crash report sent (report ID: b392cf8427a863d9)
Initializing gradle...
Crash report written to D:\MySpace\Flutter\FlutterProject\myapp1\flutter_09.log;
please let us know at https://github.com/flutter/flutter/issues.
PS D:\MySpace\Flutter\FlutterProject\myapp1>
```

**Comment by pavariss** [comment]
C:\src\flutter>flutter doctor -v
[√] Flutter (Channel stable, v1.5.4-hotfix.2, on Microsoft Windows [Version 10.0.14393], locale en-US)
    • Flutter version 1.5.4-hotfix.2 at C:\src\flutter\flutter
    • Framework revision 7a4c33425d (3 weeks ago), 2019-04-29 11:05:24 -0700
    • Engine revision 52c7a1e849
    • Dart version 2.3.0 (build 2.3.0-dev.0.5 a1668566e5)


[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at C:\Users\pavarisl\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    • All Android licenses accepted.

[√] Android Studio (version 3.4)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 35.3.1
    • Dart plugin version 183.6270
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[√] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 7.1.1 (API 25) (emulator)

• No issues found!

**Comment by pavariss** [comment]
tried run >> gradle -Dhttp.proxyHost=***  -Dhttp.proxyPort=8080 -Dhttp.proxyUser=*** -Dhttp.proxyPassword=****

not work

**Comment by pavariss** [comment]
Log


Flutter crash report; please file at https://github.com/flutter/flutter/issues.

## command

flutter run

## exception

ProcessException: ProcessException: Process "D:\MySpace\Flutter\FlutterProject\myapp1\android\gradlew.bat" exited abnormally:
Downloading https://services.gradle.org/distributions/gradle-4.10.2-all.zip


Exception in thread "main" java.net.ConnectException: Connection timed out: connect
	at java.net.DualStackPlainSocketImpl.connect0(Native Method)
	at java.net.DualStackPlainSocketImpl.socketConnect(DualStackPlainSocketImpl.java:79)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:673)
	at sun.security.ssl.BaseSSLSocketImpl.connect(BaseSSLSocketImpl.java:173)
	at sun.net.NetworkClient.doConnect(NetworkClient.java:180)
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:432)
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:527)
	at sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:264)
	at sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:367)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:191)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1138)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1032)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:177)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1546)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1474)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254)
	at org.gradle.wrapper.Download.downloadInternal(Download.java:58)
	at org.gradle.wrapper.Download.download(Download.java:44)
	at org.gradle.wrapper.Install$1.call(Install.java:61)
	at org.gradle.wrapper.Install$1.call(Install.java:48)
	at org.gradle.wrapper.ExclusiveFileAccessManager.access(ExclusiveFileAccessManager.java:65)
	at org.gradle.wrapper.Install.createDist(Install.java:48)
	at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:128)
	at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)
  Command: D:\MySpace\Flutter\FlutterProject\myapp1\android\gradlew.bat -v

```
#0      runCheckedAsync (package:flutter_tools/src/base/process.dart:255:5)
<asynchronous suspension>
#1      _initializeGradle (package:flutter_tools/src/android/gradle.dart:225:9)
<asynchronous suspension>
#2      _ensureGradle (package:flutter_tools/src/android/gradle.dart:206:37)
<asynchronous suspension>
#3      _readGradleProject (package:flutter_tools/src/android/gradle.dart:134:31)
<asynchronous suspension>
#4      _gradleProject (package:flutter_tools/src/android/gradle.dart:111:34)
<asynchronous suspension>
#5      getGradleAppOut (package:flutter_tools/src/android/gradle.dart:105:29)
<asynchronous suspension>
#6      AndroidApk.fromAndroidProject (package:flutter_tools/src/application_package.dart:154:23)
<asynchronous suspension>
#7      ApplicationPackageFactory.getPackageForPlatform (package:flutter_tools/src/application_package.dart:46:32)
<asynchronous suspension>
#8      FlutterDevice.runHot (package:flutter_tools/src/resident_runner.dart:345:56)
<asynchronous suspension>
#9      HotRunner.run (package:flutter_tools/src/run_hot.dart:253:39)
<asynchronous suspension>
#10     RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:428:37)
<asynchronous suspension>
#11     FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:559:18)
#12     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:77:64)
#13     _rootRunUnary (dart:async/zone.dart:1132:38)
#14     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#15     _FutureListener.handleValue (dart:async/future_impl.dart:126:18)
#16     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:639:45)
#17     Future._propagateToListeners (dart:async/future_impl.dart:668:32)
#18     Future._complete (dart:async/future_impl.dart:473:7)
#19     _SyncCompleter.complete (dart:async/future_impl.dart:51:12)
#20     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:28:18)
#21     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:294:13)
#22     RunCommand.usageValues (package:flutter_tools/src/commands/run.dart)
#23     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:77:64)
#24     _rootRunUnary (dart:async/zone.dart:1132:38)
#25     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#26     _FutureListener.handleValue (dart:async/future_impl.dart:126:18)
#27     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:639:45)
#28     Future._propagateToListeners (dart:async/future_impl.dart:668:32)
#29     Future._complete (dart:async/future_impl.dart:473:7)
#30     _SyncCompleter.complete (dart:async/future_impl.dart:51:12)
#31     _AsyncAwaitCompleter.complete.<anonymous closure> (dart:async-patch/async_patch.dart:33:20)
#32     _rootRun (dart:async/zone.dart:1124:13)
#33     _CustomZone.run (dart:async/zone.dart:1021:19)
#34     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:947:23)
#35     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#36     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#37     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:115:13)
#38     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:172:5)
```

## flutter doctor

```
[32m[✓][39m Flutter (Channel stable, v1.5.4-hotfix.2, on Microsoft Windows [Version 10.0.14393], locale en-US)
    [32m•[39m Flutter version 1.5.4-hotfix.2 at C:\src\flutter\flutter
    [32m•[39m Framework revision 7a4c33425d (3 weeks ago), 2019-04-29 11:05:24 -0700
    [32m•[39m Engine revision 52c7a1e849
    [32m•[39m Dart version 2.3.0 (build 2.3.0-dev.0.5 a1668566e5)

[32m[✓][39m Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    [32m•[39m Android SDK at C:\Users\pavarisl\AppData\Local\Android\sdk
    [32m•[39m Android NDK location not configured (optional; useful for native profiling support)
    [32m•[39m Platform android-28, build-tools 28.0.3
    [32m•[39m Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    [32m•[39m Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    [32m•[39m All Android licenses accepted.

[32m[✓][39m Android Studio (version 3.4)
    [32m•[39m Android Studio at C:\Program Files\Android\Android Studio
    [32m•[39m Flutter plugin version 35.3.1
    [32m•[39m Dart plugin version 183.6270
    [32m•[39m Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[32m[✓][39m Connected device (1 available)
    [32m•[39m Android SDK built for x86 • emulator-5554 • android-x86 • Android 7.1.1 (API 25) (emulator)

[32m•[39m No issues found!
```

**Comment by BondarenkoStas** [comment]
@pavariss do you still experience the issue? I think it should be already fixed in https://github.com/flutter/flutter/commit/c5e2bf5cfe9c849ef2006a8904dc48782eb02b7f at flutter  v1.10.2, can you please try it using dev or master channel?
Otherwise please try a solution from here https://github.com/flutter/flutter/issues/41353.

**Comment by pavariss** [comment]
> @pavariss do you still experience the issue? I think it should be already fixed in [c5e2bf5](https://github.com/flutter/flutter/commit/c5e2bf5cfe9c849ef2006a8904dc48782eb02b7f) at flutter v1.10.2, can you please try it using dev or master channel?
> Otherwise please try a solution from here #41327 resolved

> @pavariss do you still experience the issue? I think it should be already fixed in [c5e2bf5](https://github.com/flutter/flutter/commit/c5e2bf5cfe9c849ef2006a8904dc48782eb02b7f) at flutter v1.10.2, can you please try it using dev or master channel?
> Otherwise please try a solution from here #41353.
